docsite/docs/Docs/fal-cli/environments.md

```diff
@@ -0,0 +1,294 @@
+    type: venv
+    requirements:
+      - boto3==1.24
+      - google-cloud==0.34
+      - azure==5.0.0
+```
+
+The `interact-cloud` named environment above will have 3 dependencies (`boto3`,
+`google-cloud` and `azure` libraries, as well as their dependencies) and each
+hook/model that uses this environment can freely import any of the desired
+functions that reside under these dependencies.
+
+### Using conda-based solutions
+
+If your program depends on packages from Conda (or if you want to isolate the system
+level dependencies), you can use `conda` as the environment management backend.
+It is very similar to `venv`, but this time instead of defining a list of
+`requirements` you will define a list of `packages` from the conda channels you
+have configured.
+
+```yml
+environments:
+  - name: train
+    type: conda
+    packages:
+      - fbprophet=0.7.1
+      - scikit-learn=1.1.2
+```
+
+> Note: The environment above will be built with using the `conda` executable.
+> If there aren't any executables named `conda` under your shell's binary search
+> path, you can customize where Fal needs to look for the `conda` binary by
+> specifying the `FAL_CONDA_HOME` environment variable. E.g. if you have a
+> `miniconda` installation under `~/Downloads`, you can start fal with
+> `FAL_CONDA_HOME=~/Downloads/miniconda3/bin fal flow run`.
+
+Fal currently uses the global `~/.condarc` file when looking for channels to
+install packages (or otherwise defaulting to the `conda` executable's bundled
+search channels), so if you want to use packages from `conda-forge` (or any
+other non-default channels), you can configure them with the following commands:
+
+```console
+$ conda config --add channels conda-forge
+$ conda config --set channel_priority strict
+```
+
+## Specifying environments
+
+You can specify environments for your hooks and models under your schema file
+(`schema.yml`). There are currently two scopes to configure the environment
+option, one is in the model-level (which all the hooks would automatically
+inherit from) and the other one is in the hook-level (so only the selected hook
+would run in the specified environment, and nothing else would be affected by
+it).
+
+### Changing a hook's environment
+
+If you only want to run a specific hook on the environment, you can define the
+`environment` attribute for that hook and Fal will ensure that the hook will be
+executed under the specified environment.
+
+```yml
+version: 2
+
+models:
+  - name: reorder_alphabetically
+    meta:
+      fal:
+        pre-hook:
+          - print_enter_banner.py
+
+        post-hook:
+          - print_exit_banner.py
+          - path: send_slack_alert.py
+            environment: alerts
+```
+
+In the example above, both the model `reoder_alphabetically` and the banner printing
+scripts (`print_enter_banner.py` / `print_exit_banner.py`) will use your
+computer's local environment since they might not depend on any external Python
```

Comment:
The word "local" here might be confusing. We could say that it will use the same environment as dbt.